### PR TITLE
Block prop on first final

### PR DESCRIPTION
**Pull request by jflo**
Stops block propagation via devp2p when network finalizes, not at TTD as previous.

### CODE REVIEW

File: besu/src/main/java/org/hyperledger/besu/controller/BesuControllerBuilder.java [hunk 1]
```diff
@@ -100,6 +100,7 @@
 import java.util.OptionalLong;
 import java.util.function.Supplier;
 
+import org.jetbrains.annotations.NotNull;
```
Comment (garyschulte) [inline besu/src/main/java/org/hyperledger/besu/controller/BesuControllerBuilder.java:103]: Did you mean javax.validation.constraints.NotNull ?  the jetbrains annotation only has an effect in the IDE afaik.

File: ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/sync/AbstractBlockPropagationManagerTest.java [hunk 2]
```diff
@@ -842,25 +845,27 @@ public void shouldThrowErrorWhenNoValidPeerAvailable() {
   }
 
   @Test
-  public void shouldStopWhenTTDReached() {
+  public void shouldStopWhenFinalized() {
```
Comment (garyschulte) [inline ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/sync/AbstractBlockPropagationManagerTest.java:848]: non-blocking, but an additional test that asserts block propagation never starts if we are past TTD would be a good measure.  I don't see a DefaultSynchronizerTest class however.

**Review by garyschulte — CHANGES_REQUESTED** [review]
LGTM, minor change and non-blocking ask

**Comment by garyschulte** [comment]
FYI, running hive master against this branch resolves the 4 failing gossip engine-transition tests 👍

**Review by fab-10 — APPROVED** [review]

File: besu/src/main/java/org/hyperledger/besu/controller/BesuControllerBuilder.java (hunk 1 above)
Comment (garyschulte) [reply besu/src/main/java/org/hyperledger/besu/controller/BesuControllerBuilder.java]: removed per convo with @jflo

**Pull request merged**